### PR TITLE
directory-listing

### PR DESCRIPTION
**Pull request by bean00**
I don't really like the `if` statement in `getMediaType`, but I prefer to have that code there, so that my `buildHeaders` private method can be fully extracted out of `processRequest`. I am open to suggestions on way to deal with the `if` statement.

### CODE REVIEW

File: src/main/java/com/bean00/server/RequestProcessor.java [hunk 1]
```diff
@@ -21,34 +34,61 @@ public Response processRequest(Request request) throws IOException {
...
 
         return response;
     }
 
-    private HashMap<String, String> buildHeaders(byte[] body, String requestURL) throws IOException {
-        HashMap<String, String> headers = new HashMap<>();
+    private byte[] buildHtmlAsAByteArray() {
+        String htmlAsString = htmlBefore;
+        htmlAsString += getHtmlForListOfContents();
+        htmlAsString += htmlAfter;
```
Comment (tbaik) [inline src/main/java/com/bean00/server/RequestProcessor.java]: It should be okay to just put this all in one line, right?

File: src/main/java/com/bean00/server/RequestProcessor.java [hunk 2]
```diff
@@ -21,34 +34,61 @@ public Response processRequest(Request request) throws IOException {
...
+        byte[] body = htmlAsString.getBytes();
 
-        int contentLength = body.length;
-        String contentLengthAsString = Integer.toString(contentLength);
-        headers.put("Content-Length", contentLengthAsString);
+        return body;
+    }
+
+    private String getHtmlForListOfContents() {
```
Comment (tbaik) [inline src/main/java/com/bean00/server/RequestProcessor.java]: It's starting to look like you're giving this class more responsibilities than it should take care of

File: src/main/java/com/bean00/response/Response.java [hunk 3]
```diff
@@ -58,6 +58,13 @@ public String getHeader(String header) {
         return headers.get(header);
     }
 
+    public int getContentLength() {
```
Comment (tbaik) [inline src/main/java/com/bean00/response/Response.java]: Is this used anywhere besides tests?

File: src/main/java/com/bean00/server/RequestProcessor.java (hunk 1 above)
Comment (bean00) [reply src/main/java/com/bean00/server/RequestProcessor.java]: Yep. Done.

File: src/main/java/com/bean00/response/Response.java (hunk 3 above)
Comment (bean00) [reply src/main/java/com/bean00/response/Response.java]: I've made `getContentLength` a private helper method in the test file (for the file where I used the method multiple times)

**Review by jdesrosiers — COMMENTED** [review]
Take a look at your `DataStore` interface. Notice how coupled your names are to the file system. What will happen when you want to write a `PostgresDataStore` or a `MongoDbDataStore` or a `RedisDataStore`. Do your interface method names still make sense? Always try to use the vocabulary of the domain (HTTP in this case) when choosing names rather than using programming vocabulary. Instead of using `file` and `directory` for names, what domain specific term can you use that will make sense no mater what kind of data store it is?

File: src/main/java/com/bean00/server/RequestProcessor.java (hunk 2 above)
Comment (bean00) [reply src/main/java/com/bean00/server/RequestProcessor.java]: I've made the methods in the `FileSystemDataStore` more general, extracted the methods to build up the html body into their own class, and cleaned up the `RequestProcessor`
- *Note: I decided to throw a `NotFoundHttpException` in `processRequest` if the resource doesn't exist. I considered passing in `dataStore` to `RequestParser`s constructor, or `parseRequest`, but ultimately decided to handle the 404 error in my `RequestProcessor` (the changes were much easier to make in the latter case)

File: src/main/java/com/bean00/server/RequestParser.java [hunk 4]
```diff
@@ -31,14 +31,18 @@ public Request parseRequest() throws IOException {
 
         String requestMethod = requestWords[0];
         String requestURL = requestWords[1];
+        // TODO: check if requestURL is valid
```
Comment (jdesrosiers) [inline src/main/java/com/bean00/server/RequestParser.java]: Are you sure that is the responsibility of the `RequestParser`?

File: src/main/java/com/bean00/server/RequestProcessor.java [hunk 5]
```diff
@@ -20,35 +20,32 @@ public Response processRequest(Request request) throws IOException {
...
-        headers.put("Content-Length", contentLengthAsString);
-
-        String contentType = dataStore.getMediaType(requestURL);
+        headers.put("Content-Length", contentLength);
         headers.put("Content-Type", contentType);
 
         return headers;
     }
 
+    private String getContentLengthAsAString(byte[] body) {
```
Comment (jdesrosiers) [inline src/main/java/com/bean00/server/RequestProcessor.java]: This is a terrible method name.

Comment (bean00) [reply src/main/java/com/bean00/server/RequestProcessor.java]: Inlined this method

File: src/main/java/com/bean00/server/RequestParser.java (hunk 4 above)
Comment (bean00) [reply src/main/java/com/bean00/server/RequestParser.java]: Removed the old comments

**Review by jdesrosiers — APPROVED** [review]

**Pull request merged**